### PR TITLE
BERT fine-tuning with huggingface transformers

### PR DESCRIPTION
**Pull request by BenjaminBossan**
Add a notebook that shows what steps need to be taken to fine-tune a
BERT model in skorch with the help of the [huggingface transformers
library](https://huggingface.co/docs/transformers/index).

Huggingface transformers is probably the most popular library that
implements BERT and other similar models. Ideally, skorch can work with
transformers with as little friction as possible.

There are actually surprisingly few steps involved, but some of them may
not be obvious to most users, which is why the notebook could be
helpful.

(One minor annoyance being explained in #838)

As a cherry on top, the notebook also shows mixed precision training
with huggingface accelerate, which indeed results in a very nice boost
of training and prediction speed.

Other than that, the notebook is actually quite sparse in explanations,
instead referring to the [fine-tuning docs](https://huggingface.co/docs/transformers/training) of hugginface itself.

[Notebook link](https://github.com/skorch-dev/skorch/blob/notebook-huggingface-transformers-fine-tuning/notebooks/Hugging_Face_Finetuning.ipynb)

Note:

I accidentally pushed this to master first. The change was reverted
immediately, my apologies for this. I should maybe not do this kind
of work at midnight.

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
Note: This should wait for the results of #841. If that PR is merged, I will rework this one to make use of the new transformers.

**Comment by BenjaminBossan** [comment]
@ottonemo @thomasjpfan After merging the Hugging Face transformer PR, I updated this [notebook](https://github.com/skorch-dev/skorch/blob/notebook-huggingface-transformers-fine-tuning/notebooks/Hugging_Face_Finetuning.ipynb) to use it, thus making it simpler.

I also crammed in some fixes to the docs (sorry for not creating a separate PR). New docs can be found [here](https://skorch.readthedocs.io/en/notebook-huggingface-transformers-fine-tuning/index.html).

File: notebooks/Hugging_Face_Finetuning.ipynb [hunk 1]
```diff
@@ -0,0 +1,920 @@
+{
```
Comment (thomasjpfan) [inline notebooks/Hugging_Face_Finetuning.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # make to choose a tokenizer and bert model that work together
</pre><p>Is the following better wording?</p><p> </p><blockquote>We choose a tokenizer and bert model that works together</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/skorch-dev/skorch/pull/839/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (thomasjpfan) [inline notebooks/Hugging_Face_Finetuning.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    TOKENIZER = "bert-base-uncased"
</pre><p>Should we the distilbert-base-uncased tokenizer here?</p><p> </p><p>XREF: <a href="https://huggingface.co/distilbert-base-uncased" target="_blank">https://huggingface.co/distilbert-base-uncased</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/skorch-dev/skorch/pull/839/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (BenjaminBossan) [reply notebooks/Hugging_Face_Finetuning.ipynb:1]: Changed.

**Review by thomasjpfan — APPROVED** [review]

**Pull request merged**